### PR TITLE
Add `filetype` to NYPL script

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #536 by @obulat

## Description

This PR adds `filetype` to the NYPL script. It is not possible to get `filesize` without actually downloading the image.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_catalog/dags/providers/provider_api_scripts/nypl.py [hunk 1]
```diff
@@ -155,31 +160,54 @@ def _get_creators(creatorinfo):
...
-    image_url = _get_preferred_image(image_type, image_url_dimensions)
-
-    return image_url
-
-
-def _get_preferred_image(image_type, dimension_list):
+    image_type = {
+        parse_qs(urlparse(img.get("$")).query)["t"][0]: {
+            "url": img.get("$"),
+            "description": img.get("description"),
```
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/nypl.py:195]: Using `dict::get` here makes me think it's possible that some of those values might not exist in the dictionary. But if not, then other aspects of this comprehension (e.g. `parse_qs(urlparse(None).query)["t"]`) will fail. We can probably simplify this to:
```suggestion
        parse_qs(urlparse(img["$"]).query)["t"][0]: {
            "url": img["$"],
            "description": img["description"],
```

**Review by AetherUnbound — APPROVED** [review]
Looks good! One suggested change 🙂

File: openverse_catalog/dags/providers/provider_api_scripts/nypl.py [hunk 2]
```diff
@@ -155,31 +160,54 @@ def _get_creators(creatorinfo):
...
-    if image_url_dimensions is None:
-        image_url_dimensions = IMAGE_URL_DIMENSIONS
-    image_type = {
-        parse_qs(urlparse(img.get("$")).query)["t"][0]: img.get("$") for img in images
+def _get_filetype(description: str):
+    """
+    Extracts the filetype from a description string like:
+    "Cropped .jpeg (1600 pixels on the long side)"
+    :param description: the description string
+    :return:  jpeg | gif
```
Comment (rwidom) [inline openverse_catalog/dags/providers/provider_api_scripts/nypl.py:173]: That's so intense that there isn't just a file name extension in a url somewhere. [Right?](https://digitalcollections.nypl.org/items/5d399174-5bd8-4e97-95ad-a65c8204d892) How did you decide to limit the regex to jpeg and gif? (e.g. and not jpg or png) Also, how is the file type data used in the front end?

Comment (obulat) [reply openverse_catalog/dags/providers/provider_api_scripts/nypl.py:173]: Yes, such URLs really pose a problem when adding data. This happens more often for audio files than for audio.

But NYPL is actually not that bad: it only has `jpeg`s and `gif`s (and very huge original `tiff`s that we do not use). `gif` is usually used for a very small version of the image, so it's actually mostly `jpeg`.

File type data is used for filtering the images by extension, both in the API and in the front end. File type also should be displayed on the individual result pages.

**Comment by obulat** [comment]
Closing this PR in favor of #630.

**Pull request closed without merging**